### PR TITLE
fix: Support fetching multiple databases and schemas and their associated tables

### PR DESCRIPTION
**Pull request by guenp**
Picking this up from previous PRs by @Mause and @Alex-Monahan (see https://github.com/Mause/duckdb_engine/pull/748 and https://github.com/Mause/duckdb_engine/pull/740)

Due to having multiple databases in DuckDB, the Postgres dialect of get_schema_names has duplicates (Ex: one copy of "main" for every database, including temp and system).

This change adds a method `get_schema_names` that returns a list of schema names formatted as `<db name>.<schema name>` and lets you get a schema's associated tables using `get_table_names`.

### CODE REVIEW

**Comment by Mause** [comment]
Are you able to look at the failing checks?

**Comment by guenp** [comment]
@Mause, sure thing, I think all checks are passing now.

File: duckdb_engine/tests/test_basic.py [hunk 1]
```diff
@@ -166,6 +166,51 @@ def test_get_tables(inspector: Inspector) -> None:
...
+    # Run each command in a separate transaction
+    for cmd in [
+        """CREATE SCHEMA "quack quack" """,
+        """ATTACH ':memory:' AS "daffy duck" """,
+        """CREATE SCHEMA "daffy duck"."quack quack" """,
+        """CREATE TABLE "daffy duck"."quack quack"."t1" (i INTEGER, j INTEGER);""",
+        """CREATE TABLE "daffy duck"."quack quack"."t2" (i INTEGER, j INTEGER);""",
+        """CREATE SCHEMA "daffy duck"."you're "" despicable" """,
+    ]:
+        with engine.connect() as conn:
```
Comment (Mause) [inline duckdb_engine/tests/test_basic.py:183]: Isn't this just creating a new in memory database for every loop iteration?

Comment (guenp) [reply duckdb_engine/tests/test_basic.py:183]: Ah, you are right, It does indeed look like [closing](https://duckdb.org/docs/api/c/connect.html#duckdb_close) the connection deallocates the memory. My bad. The test passes for me locally so will try to repro this first. Thanks

**Comment by guenp** [comment]
Alright, I was able to repro the test failure locally and I think I fixed it now! Just to be 100% sure I ran the CI pipeline here: https://github.com/motherduckdb/duckdb_engine/actions/runs/7118602752.

**Comment by Mause** [comment]
Thanks! This is now released in 0.9.4

**Pull request merged**